guide_drafts/trait_derives.md

```diff
@@ -93,96 +104,177 @@ fn main() {
     pub id: i32,
+    pub email: String,
+}
+
+#[derive(Debug, QueryableByName)]
+pub struct UserName {
+    #[sql_type="Text"]
     pub first_name: String,
+    #[sql_type="Text"]
     pub last_name: String,
-    pub email: String,
 }
 
-impl<ST, DB> Queryable<ST, DB> for User where
-    DB: Backend + HasSqlType<ST>,
-    (i32, String, String, String): FromSqlRow<ST, DB> {
-    
-    type Row = (i32, String, String, String);
-
-    fn build(row: &Self::Row) -> Self {
-        User {
-            id: row.0,
-            first_name: row.1,
-            last_name: row.2,
-            email: row.3,
-        }
-    }
+#[derive(Insertable)]
+#[table_name="users"]
+pub struct NewUser<'a> {
+    pub first_name: &'a str,
+    pub last_name: &'a str,
+    pub email: &'a str,
+ }
+
+```
+
+```rust
+// File: src/main.rs
+
+use diesel::prelude::*;
+
+fn main() {
+  let new_user = NewUser { 
+      first_name: "Gordon", 
+      last_name: "Freeman", 
+      email: "gordon.freeman@blackmesa.co" 
+  };
+
+  diesel::insert_into(users::table)
+      .values(&new_user)
+      .execute(&db_connection)
+      .expect("Error inserting row into database");
+
+  let users_emails = sql_query("SELECT users.id, users.email FROM users ORDER BY id")
+    .load::<UserEmail>(&connection);
+  println!("{:?}", users_emails); 
+  // > User { id: 1, email: "gordon.freeman@blackmesa.co" }
+  
+  let users_name = sql_query("SELECT users.first_name, users.last_name FROM users ORDER BY id")
+    .load::<UserName>(&connection);
+  println!("{:?}", users_name); 
+  // > User { first_name: "Gordon", last_name: "freeman" }
 }
 ```
 
-The advantages of using the derive annotation should be clear here, as the more columns you have, the more code you would have to write.
+If we were to forget one of `QueryableByName`'s required struct annotations,
+we would see the following error.
+
+```rust
+error: Your struct must either be annotated with `#[table_name = "foo"]` or have all of its fields annotated with `#[sql_type = "Integer"]`
+ --> src/models.rs:4:17
+  |
+4 | #[derive(Debug, QueryableByName)]
+  |                 ^^^^^^^^^^^^^^^
+```
 
 ## <a name="insertable"></a>Insertable
 
-Adding the [`#[derive(Insertable)]`][insertable_doc] trait annotation to your model struct allows records to be inserted into the database via that struct.
-You will need `Insertable` when using the `diesel::insert(&my_new_row).into(my_table::table)` API.
-You may pass a reference to a single struct to insert a single record. Pass a Vec or a slice to insert multiple records at once.
+The `Insertable` trait is meant to be implemented on structs for 
+the purpose of inserting row data into a table.
+To implement `Insertable` on your struct,
+add the [`#[derive(Insertable)]`][insertable_doc] trait annotation.
+
+This implementation will enable you to use the `insert_into(my_table::table)` 
+method to generate a `SQL INSERT` statement for that table.
+You may chain `values(&row_values)` or `default_values()`
```

Comment:
Do you think we should put a dot in front of these to indicate that they're a method? Should these be links?